runtime/native/rt_bignum_parse.c

```
@@ -0,0 +1,275 @@
+    } else if (data[start] == '-') {
+        neg = true;
+        start++;
+    }
+    if (start >= end) {
+        return BN_ERR_NEG_SHIFT;
+    }
+    size_t i = start;
+    uint8_t* digits = (uint8_t*)malloc(len + 1);
+    size_t digits_len = 0;
+    while (i < end && data[i] >= '0' && data[i] <= '9') {
+        digits[digits_len++] = data[i];
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)